owncloud/core
# Rewire notify by email

## Description
- [x] Get rid of legacy `API::register` invocations in favor of routing
- [x]  Kill an ugly OCSShareWrapper
- [x] Refactor `Share20OCS` to be a controller
- [x] Move 2 endpoints from **core/ajax/share.php** into the new controller
- [x] Rewrite the implementation for these endpoints in order not to use legacy sharing methods

## Related Issue
- Fixes https://github.com/owncloud/core/issues/32494
- unblocks https://github.com/owncloud/core/pull/26608

## Motivation and Context
Sharing code cleanup

## How Has This Been Tested?
1. share with user
2. click `notify by email`
3. check that email has been arrived
4. reload the page and check that there is no  `notify by email` button any more
5. Login as user and check that the emailed permalink works

1. share with group
2. click `notify by email`
3. check that email for all users with emails have been arrived
4. reload the page and check that there is no  `notify by email` button any more
5. Login as any user from this group and check that permalink  works (TBH I didn't test them all) 

## Types of changes

## Checklist:

## Open tasks:
Since it doesn't beak anything a backport would be nice
